pkgs/data/themes/catppuccin-kde/default.nix

```diff
@@ -2,23 +2,54 @@
 , stdenvNoCC
 , fetchFromGitHub
 , fetchpatch
-, flavour ? [ "frappe" ]
-, accents ? [ "blue" ]
-, winDecStyles ? [ "modern" ]
+, writeShellApplication
+, lightly-plasma
+, gnutar
+, unzip
+, flavour ? "frappe"
+, accent ? "blue"
+, winDecStyle ? "modern"
+, withLightly ? true
 }:
 
 let
-  validFlavours = [ "mocha" "macchiato" "frappe" "latte" ];
-  validAccents = [ "rosewater" "flamingo" "pink" "mauve" "red" "maroon" "peach" "yellow" "green" "teal" "sky" "sapphire" "blue" "lavender" ];
-  validWinDecStyles = [ "modern" "classic" ];
+  # The ordering of these options is important; they must match the indicies in
+  # the installation script.
+  #
+  # https://github.com/catppuccin/kde/blob/main/install.sh
+  indicies = builtins.mapAttrs (_: values: builtins.listToAttrs (lib.imap1 (lib.flip lib.nameValuePair) values)) {
```

Comment:
Nitpick: this is a bit hard to read IMO. Either use a let-in to inherit these functions, and/or use `lib.pipe` to reduce nesting. Also, it's spelled 'indices':
```nix
let
  inherit (builtins) mapAttrs listToAttrs;
  inherit (lib) pipe imap1 flip nameValuePair;

  indices = mapAttrs (_: flip pipe [
    (imap1 (flip nameValuePair))
    listToAttrs
  ]) {
    # ...
  };
in # ...
```